parachain/runtime/src/lib.rs

```diff
@@ -235,6 +238,64 @@ impl pallet_utility::Config for Runtime {
     type WeightInfo = ();
 }
 
+pub struct RelayChainBlockNumberProvider<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: cumulus_pallet_parachain_system::Config> BlockNumberProvider for RelayChainBlockNumberProvider<T> {
+    type BlockNumber = BlockNumber;
+
+    fn current_block_number() -> Self::BlockNumber {
+        cumulus_pallet_parachain_system::Pallet::<T>::validation_data()
+            .map(|d| d.relay_parent_number)
+            .unwrap_or_default()
+    }
+}
+
+parameter_types! {
+    pub MinVestedTransfer: Balance = 0;
+    pub const MaxVestingSchedules: u32 = 100;
+}
+
+parameter_types! {
+    pub KintsugiLabsAccounts: Vec<AccountId> = vec![
+        hex_literal::hex!["249cf21ac84a06f2e1661e215e404530529f3932034abe9a5b8e3da5eee8b374"].into(),	// 5CtiCmFoHDSiLLoaBkqj9sGonciXgAvj2mnbZ8DZe4bpLLQ7
+        hex_literal::hex!["4aa5c577b4dcfcd72e78a728ca52707eed424b7bfa6c584b3ad9caa8087bdd20"].into(),	// 5DkaeN4Rpq4cfyExfopCSEDiJAEpKR8A4szT398p5271bVaa
+        hex_literal::hex!["accb25b6794d8efa88397ccc05017727f658494484525ae8a3bd4c0bc0316e16"].into(),	// 5FyGSWj5b6VcK7L9psiDE6RQBp1XC3SzbXNt5TU1Zqr56QGY
+    ];
+}
+
+pub struct EnsureKintsugiLabs;
+impl EnsureOrigin<Origin> for EnsureKintsugiLabs {
+    type Success = AccountId;
+
+    fn try_origin(o: Origin) -> Result<Self::Success, Origin> {
+        Into::<Result<RawOrigin<AccountId>, Origin>>::into(o).and_then(|o| match o {
+            RawOrigin::Signed(caller) => {
+                if KintsugiLabsAccounts::get().contains(&caller) {
+                    Ok(caller)
+                } else {
+                    Err(Origin::from(Some(caller)))
+                }
+            }
+            r => Err(Origin::from(r)),
+        })
+    }
+
+    #[cfg(feature = "runtime-benchmarks")]
+    fn successful_origin() -> Origin {
+        Origin::from(RawOrigin::Signed(Default::default()))
+    }
+}
+
+impl orml_vesting::Config for Runtime {
```

Comment:
The vesting is block-based rather than time-based. You raised this earlier as a point of attention. Was it confirmed that block-based is ok, from a legal point of view?

Reply:
The point was that the total rewards may not divide cleanly into the number of blocks produced per year (since we cannot be sure an exact amount will be produced).